src/frontend/Debug_data_generation.ml

```diff
@@ -89,8 +89,65 @@ let gen_int m t = wrap_int (gen_num_int m t)
 let gen_real m t = wrap_real (gen_num_real m t)
 
 let gen_row_vector m n t =
-  { expr= RowVectorExpr (repeat_th n (fun _ -> gen_real m t))
-  ; emeta= {loc= Location_span.empty; ad_level= DataOnly; type_= UMatrix} }
+  let extract_var e =
+    match e with {expr= Variable x; _} -> Map.find_exn m x.name | _ -> e
+  in
+  let gen_bounded t e =
+    match e with
+    | {expr= RowVectorExpr unpacked_e; _} | {expr= ArrayExpr unpacked_e; _} ->
+        wrap_row_vector (List.map ~f:(fun x -> gen_real m (t x)) unpacked_e)
+    | _ ->
+        failwith
+          ( "Bad bounded (upper OR lower) expr: "
+          ^ Sexp.to_string [%sexp (e : (typed_expr_meta, fun_kind) expr_with)]
+          )
```

Comment:
I recently discovered a better alternative to `failwith (Sexp.to_string _)`
```ocaml
raise_s [%message "Bad bounded (upper OR lower) expr: " (e : (typed_expr_meta, fun_kind) expr_with)]
```
It's both shorter and formats the s-expression instead of cramming it all on a single line.